test-common/utils.c

```diff
@@ -1846,6 +1883,266 @@ get_update_server_dir (GFile *client_root)
+                                                            "repo",
+                                                            NULL);
+  g_autoptr(GFile) repo_directory = g_file_new_for_path (repo_directory_path);
+  g_autofree gchar *runtime_directory = g_build_filename (g_file_get_path(flatpak_build_directory_path),
+                                                          "runtime",
+                                                          NULL);
+  const gchar **flatpak_name_iter = NULL;
+
+  if (!g_file_make_directory_with_parents (flatpak_build_directory_path, NULL, error))
+    return FALSE;
+
+  /* We need to set the collection-id on both the remote end (the repo
+   * are pulling from) and in the remote configuration in the local mirror
+   * below */
+  if (!flatpak_populate_runtime (updater_dir,
+                                 runtime_directory_path,
+                                 repo_directory_path,
+                                 "org.test.Runtime",
+                                 collection_id,
+                                 error))
+    return FALSE;
+
+  if (!flatpak_remote_add (updater_dir,
+                           repo_name,
+                           repo_directory_path,
+                           error))
+    return FALSE;
+
+  if (!flatpak_install (updater_dir,
+                        "test-repo",
+                        "org.test.Runtime",
+                        error))
+    return FALSE;
+
+
+  /* Now that we have our runtime installed, lets go ahead and build and export
+   * all the apps flatpak_name_iter our repo */
+  for (flatpak_name_iter = flatpak_names; *flatpak_name_iter != NULL; ++flatpak_name_iter)
+    {
+      g_autofree gchar *app_dir = g_build_filename (apps_directory,
+                                                    *flatpak_name_iter,
+                                                    NULL);
+      g_autoptr(GFile) app_path = g_file_new_for_path (app_dir);
+
+      if (!flatpak_populate_app (updater_dir,
+                                 app_path,
+                                 *flatpak_name_iter,
+                                 "org.test.Runtime",
+                                 repo_directory_path,
+                                 error))
+        return FALSE;
+    }
+
+  /* Now that we have our runtime installed, lets go ahead and build and export
+   * all the apps into our repo */
+  for (flatpak_name_iter = preinstall_flatpak_names; *flatpak_name_iter != NULL; ++flatpak_name_iter)
+    {
+      if (!flatpak_install (updater_dir,
+                            repo_name,
+                            *flatpak_name_iter,
+                            error))
+        return FALSE;
+    }
+
+  /* It seems like calling ostree config set will turn
+   * GPG verification back on for the repo, so the remote
+   * collection-id needs to be set after the flatpak is installed */
```

Comment:
Yeah, these tests really should be using GPG verification, since that’s what real systems will be using. I don’t want to block merging this on having GPG support in the test suite, but could you please file a follow-up task about adding it?

I’ll change this comment to include a `FIXME`.